docs/HowToCertificatesRotation.md

```diff
@@ -0,0 +1,473 @@
+    }
+  }
+}
+```
+
+## NuoDB TLS Key Rotation
+
+To renew the domain key pair certificates and the server keystore, you can either create new keystore files and certificates on your own or create them using NuoDB commands.
+All examples below will be using a helper pod started with NuoDB image and `nuocmd` command line tools.
+
+```bash
+PASSWD=changeMe
+mkdir /tmp/keys
+
+kubectl run generate-nuodb-certs \
+  --image nuodb/nuodb-ce \
+  --env="PASSWD=changeMe" \
+  --command -- 'tail' '-f' '/dev/null'
+
+kubectl exec -ti generate-nuodb-certs -- \
+  mkdir -p /tmp/keys
+```
+
+### Rotate CA Certificate
+
+Generate new CA certificate:
+
+```bash
+kubectl exec -ti generate-nuodb-certs -- \
+  nuocmd create keypair \
+    --keystore /tmp/keys/ca.p12 --store-password "$PASSWD" \
+    --dname "CN=ca.nuodb.com, OU=Eng, O=NuoDB, L=Boston, ST=MA, C=US, SERIALNUMBER=67890" \
+    --validity 36500 --ca
+
+kubectl exec -ti generate-nuodb-certs -- bash -c \
+  'nuocmd show certificate \
+    --keystore /tmp/keys/ca.p12 --store-password "$PASSWD" \
+    --cert-only > /tmp/keys/ca.cert'
+```
+
+If the CA certificate is owned by a public CA, then follow the official steps to obtain the new certificate.
+
+The newly created keystore files can be copied on the client machine and then on some AP pod.
+
+```bash
+kubectl cp generate-nuodb-certs:/tmp/keys/. /tmp/keys
+
+kubectl cp /tmp/keys/ca.cert nuodb/admin-nuodb-cluster0-0:/tmp/ca.cert
+```
+
+Add the new CA certificate to the truststore of all admin and database processes:
+
+```bash
+kubectl exec -ti admin-nuodb-cluster0-0 -- \
+  nuocmd add trusted-certificate \
+    --alias ca_prime --cert /tmp/ca.cert --timeout 30
+```
+
+The `--timeout` argument specifies the amount of time to wait for the new certificate to be propagated to the truststore of every process in the domain.
+
+Ensure that the new CA certificate with alias `ca_prime` is trusted by all APs and database processes as described in [Verify Domain Certificates](#verify-domain-certificates).
+
+Add the new CA certificate to the truststore of every SQL client.
+
+Generate and sign the new server certificates using the new CA certificate chain as described in [Rotate Server Certificate](#rotate-server-certificate).
+
+### Rotate Server Certificate
+
+Generate a new server key pair:
+
+```bash
+kubectl exec -ti generate-nuodb-certs -- \
+  nuocmd create keypair \
+    --keystore /tmp/keys/nuoadmin.p12 --store-password "$PASSWD" --ca \
+    --dname "CN=*.nuodb.svc.cluster.local, OU=Eng, O=NuoDB, L=Boston, ST=MA, C=US, SERIALNUMBER=67890"
+```
+
+In the example above, we specify the `SERIALNUMBER` field so that the new certificate has a different distinguished name from the current certificate.
+
+The new server certificates should be signed by the CA certificate.
+If the TLS keys have been generated using the `setup-keys.sh` script, the private/public key pair of the CA are available in the `ca.p12` file.
```

Comment:
Should we comment about storing the ca.p12 file securely here, since it can be used to sign new certificates, it shouldn't be left around in the _generate-nuodb-certs_ pod, locally, or copied to all NuoDB pods?

Reply:
Good point!